sys/xtimer/xtimer_core.c

```diff
@@ -480,13 +479,18 @@ static void _timer_callback(void)
 
         /* fire timer */
         _shoot(timer);
+
+        /* time update after executing callback */
+        if (!timer->trivial_callback) {
+          now = _xtimer_lltimer_now();
```

Comment:
assume you have 10 timers with non trivial callbacks, this will result in 10x `_xtimer_lltimer_now()` instead of just once with the current implementation.